confgenerator/self_logs.go

```diff
@@ -0,0 +1,138 @@
+}
+
+func generateHealthChecksLogsComponents(ctx context.Context) []fluentbit.Component {
+	out := make([]fluentbit.Component, 0)
+	out = append(out, LoggingReceiverFilesMixin{
+		IncludePaths:   []string{healthChecksLogsPath()},
+		BufferInMemory: true,
+	}.Components(ctx, healthLogsTag)...)
+	out = append(out, LoggingProcessorParseJson{
+		// TODO(b/282754149): Remove TimeKey and TimeFormat when feature gets implemented.
+		ParserShared: ParserShared{
+			TimeKey:    "time",
+			TimeFormat: "%Y-%m-%dT%H:%M:%S%z",
+		},
+	}.Components(ctx, healthLogsTag, "health-checks-json")...)
+	out = append(out, []fluentbit.Component{
+		// This is used to exclude any previous content of the health-checks file that
+		// does not contain the `severity` field.
+		{
+			Kind: "FILTER",
+			Config: map[string]string{
+				"Name":  "grep",
+				"Match": healthLogsTag,
+				"Regex": "severity INFO|ERROR|WARNING|DEBUG|DEFAULT",
+			},
+		},
+		{
+			Kind: "FILTER",
+			OrderedConfig: [][2]string{
+				{"Name", "modify"},
+				{"Match", healthLogsTag},
+				{"Rename", fmt.Sprintf("severity %s", severityKey)},
+				{"Rename", fmt.Sprintf("sourceLocation %s", sourceLocationKey)},
+			},
+		},
+	}...)
+
+	return out
+}
+
+func generateFluentBitSelfLogsComponents(ctx context.Context) []fluentbit.Component {
+	out := make([]fluentbit.Component, 0)
+	out = append(out, LoggingReceiverFilesMixin{
+		IncludePaths: []string{fluentbitSelfLogsPath(platform.FromContext(ctx))},
+		//Following: b/226668416 temporarily set storage.type to "memory"
+		//to prevent chunk corruption errors
+		BufferInMemory: true,
+	}.Components(ctx, fluentBitSelfLogTag)...)
+	out = append(out, LoggingProcessorParseRegex{
+		Regex:       `(?<message>\[[ ]*(?<time>\d+\/\d+\/\d+ \d+:\d+:\d+)] \[[ ]*(?<severity>[a-z]+)\].*)`,
+		PreserveKey: true,
+		ParserShared: ParserShared{
+			TimeKey:    "time",
+			TimeFormat: "%Y/%m/%d %H:%M:%S",
+			Types: map[string]string{
+				"severity": "string",
+			},
+		},
+	}.Components(ctx, fluentBitSelfLogTag, "self-logs-severity")...)
+	out = append(out, fluentbit.TranslationComponents(fluentBitSelfLogTag, "severity", severityKey, true,
+		[]struct{ SrcVal, DestVal string }{
+			{"debug", "DEBUG"},
+			{"error", "ERROR"},
+			{"info", "INFO"},
+			{"warn", "WARNING"},
+		})...,
+	)
+	return out
+}
+
+func structuredHealthLogsLabels() string {
+	labels := []string{
+		fmt.Sprintf("%s=%s", agentKindKey, "ops-agent"),
+		fmt.Sprintf("%s=%s", agentVersionKey, version.Version),
+		fmt.Sprintf("%s=%s", schemaVersionKey, "v1"),
+	}
+	return strings.Join(labels, ",")
+}
+
+func generateSelfLogsComponents(ctx context.Context, userAgent string) []fluentbit.Component {
+	out := make([]fluentbit.Component, 0)
+
+	out = append(out, generateFluentBitSelfLogsComponents(ctx)...)
+	out = append(out, generateHealthChecksLogsComponents(ctx)...)
+
+	healthLabels := structuredHealthLogsLabels()
+	out = append(out, stackdriverOutputComponent(healthLogsTag, userAgent, "", healthLabels))
```

Comment:
minor nit: we don't need the intermediate variable here

Reply:
Remove the addition of labels from the output plugin.